webview.h

```diff
@@ -848,24 +848,145 @@ using browser_engine = detail::cocoa_wkwebview_engine;
 //
 
 #define WIN32_LEAN_AND_MEAN
+#include <shellscalingapi.h>
```

Comment:
It seems this header is unused because we find symbols dynamically?

Reply:
Types from this header are used in the new code.

Reply:
IIRC, one of the tasks of this PR is to lower compatibility requirements.
If only `DPI_AWARENESS_CONTEXT_*` are used maybe better to define them directly to avoid inclusion of this header, because it requires recent Windows SDK, it can help to compile library with old MinGW distros, etc.

Reply:
Being able to expand compiler support would indeed be a great thing for another PR, and I would love to see some more research around this.

The primary purpose of this PR is runtime compatibility and is not intended to address any existing compile-time compatibility issues for MinGW. A nice side effect of this PR is that it lowers compile-time requirements but it is not meant to fully address this issue.

The new header (`shellscalingapi.h`) applies to Windows 8.1 so redefining types from that header will not change the SDK requirements unless we also redefine other types specific for Windows 10. This is doable but first I would like to see evidence that a modern version of MinGW actually benefits from doing so. I think it is reasonable to have stricter requirements for the SDK.

In any case, I have a couple points that I think are important to consider:

* We require a compiler that supports C++17 when compiling for Windows, thanks to Microsoft WebView2, so the compiler (e.g. MinGW) can't be too old.
* Does Microsoft WebView2 support any compiler other than MSVC?

Reply:
`SetProcessDPIAware` is located in `winuser.h` (through `windows.h`) and became available in Windows Vista.

`PROCESS_DPI_AWARENESS` / `SetProcessDpiAwareness` is located in `shellscalingapi.h` and became available in Windows 8.1.

`DPI_AWARENESS_CONTEXT` / `SetProcessDpiAwarenessContext` is located in `winuser.h` (through `windows.h`) and became available in Windows 10.